fairscale/optim/oss.py

```diff
@@ -521,7 +523,7 @@ def _broadcast_params(self) -> None:
 
                     for param in params:
                         # Bucket broadcast
-                        if self.should_bucket_param[param]:
+                        if self.bucket_size > 0 and self.should_bucket_param[param]:
```

Comment:
faster to compare against an int than to look up in the hash table/dict. Coincidentally this "fixes" https://github.com/huggingface/transformers/issues/9156#issuecomment-747146404 on single node, since this removes any dependency on bucketing